src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,6 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
+{
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
+  TH1F*  hProvdNdESignal[nFiles];
+  TH1F*  hProvdNdESignalTemp;
+  TH1F*  linCHdNdESignal;
+
+  Int_t status      = 0;
+  Int_t nBins       = 0;
+  Double_t lowEdge  = 0.0;
+  Double_t highEdge = 0.0;
+  // open file and look for histograms
+  for(Int_t iFile=0;iFile<nFiles;iFile++)
+    {
+      dNdESignalFile[iFile]  = new TFile(filenames[iFile]);
+      hProvdNdESignalTemp = (TH1F*) dNdESignalFile[iFile]->Get("hdNdE");
+      // check that dN/dE histograms are found
+      if(!hProvdNdESignalTemp) status=1;
+      // store the histogram binning from the first historgram
+      if(nBins==0)
+        {
+          nBins = hProvdNdESignalTemp->GetNbinsX();
+          lowEdge = hProvdNdESignalTemp->GetBinLowEdge(1);
+          highEdge = hProvdNdESignalTemp->GetBinLowEdge(nBins+1);
+        }
+      // check that the following histograms have the same binning than the first histogram
+      if(hProvdNdESignalTemp->GetNbinsX()!=nBins || hProvdNdESignalTemp->GetBinLowEdge(1)!=lowEdge || hProvdNdESignalTemp->GetBinLowEdge(nBins+1)!=highEdge)
+        {
+          cout << "Iact1dUnbinnedLkl::ReaddNdESignal, histogram read from file " << filenames[iFile] << " has " << hProvdNdESignalTemp->GetNbinsX() << " bins (should be " << nBins << " bins), lowEdge = " << hProvdNdESignalTemp->GetBinLowEdge(1) << " (should be " << lowEdge << ") and highEdge = " << hProvdNdESignalTemp->GetBinLowEdge(nBins+1) << " (should be " << highEdge << ")" << endl;
+        }
+      // interpolate if historgram binning divers from the binning of the first histogram
+      if(hProvdNdESignal[iFile]) delete hProvdNdESignal[iFile];
+      hProvdNdESignal[iFile] = new TH1F("hProvdNdESignal","dN/dE for signal events",nBins,lowEdge,highEdge);
+      hProvdNdESignal[iFile]->SetDirectory(0);
+      readAndInterpolate(hProvdNdESignalTemp,hProvdNdESignal[iFile]);
+      delete hProvdNdESignalTemp;
+    }
+
+  if(status==0)
+    {
+      // create histogram for the linear combination of dNdESignal
+      linCHdNdESignal = new TH1F("linCHdNdESignal","dN/dE for signal events",nBins,lowEdge,highEdge);
+      // perform the linear combination
+      for(Int_t iFile=0;iFile<nFiles;iFile++)
+        {
+          linCHdNdESignal->Add(hProvdNdESignal[iFile],branchingRatios[iFile]);
```

Comment:
No check of whether BRs sum is 1 (or at least less than 1), should be included. Maybe you want to consider normalizing BRs to sum 1? A comment about what you do with this should be included in the comment at the top of the class definition.

Reply:
Good point! I will add a check!
Does the normalizing of BRs is linked to the comment above? Do you mean, if we can't find a histogram (ie with BR 0.3), we should normalize the rest 0.7 to 1?

Reply:
No,  if **any** of the requested histos is not found, then you should return the _unsuccess_ code (1, yes? right, 1 ;))

Reply:
Yes :D